clients/src/test/java/org/apache/kafka/clients/ClusterConnectionStatesTest.java

```diff
@@ -321,4 +325,37 @@ public void testIsPreparingConnection() {
         connectionStates.disconnected(nodeId1, time.milliseconds());
         assertFalse(connectionStates.isPreparingConnection(nodeId1));
     }
+
+    @Test
+    public void testExponentialConnectionSetupTimeout() {
+        assertTrue(connectionStates.canConnect(nodeId1, time.milliseconds()));
+
+        // Check the exponential timeout growth
+        for (int n = 0; n <= Math.log((double) connectionSetupTimeoutMaxMs / connectionSetupTimeoutMs) / Math.log(2); n++) {
```

Comment:
nit: Could we define a local constant for the base `2` and reuse it everywhere?

Reply:
Yes. I'll create a constant to make the test robust.